test/HydratorAwareTraitTest.php

```diff
@@ -15,6 +15,7 @@ class HydratorAwareTraitTest extends TestCase
 {
     public function testSetHydrator(): void
     {
+        /** @psalm-suppress InvalidScalarArgument False positive */
```

Comment:
Why are we suppressing this? Isn't `HydratorAwareTrait` a valid trait? :thinking:

Reply:
This one was odd.

PHPUnit 9.5 changes the parameter type for the `$traitName` argument to `trait-string`. `HydratorAwareTrait` **is** a trait, and the `::class` usage would evaluate to its fully qualified name, which means it's a trait-string. I think that Psalm just isn't detecting that.

Reply:
Alternately, the trait-string type just is not understood in the version of Psalm we're using, so it sees `trait-string` and thinks this is a class type or similar, and, either way, cannot resolve it. I'm not sure what version of Psalm that was added.

Reply:
> trait-string type just is not understood in the version of Psalm we're using

very likely: can we try an upgrade?

Reply:
So, we're actually at the latest stable release (4.10.0). I'd qualify this as a Psalm bug.

Reply:
Ok, was worth a shot 👍